### PR TITLE
Ea 2577 solr indexing

### PR DESCRIPTION
**Pull request by SrdjanStevanetic**
Please review the code.

### CODE REVIEW

File: entity-management-definitions/src/main/java/eu/europeana/entitymanagement/definitions/exceptions/EntityCreationException.java [hunk 1]
```diff
@@ -32,8 +32,8 @@ public boolean logStacktrace() {
...
     }
 
-    @Override
-    public HttpStatus getResponseStatus() {
-        return HttpStatus.BAD_REQUEST;
-    }
+//    @Override
+//    public HttpStatus getResponseStatus() {
+//        return HttpStatus.BAD_REQUEST;
+//    }
```
Comment (ikattey) [inline entity-management-definitions/src/main/java/eu/europeana/entitymanagement/definitions/exceptions/EntityCreationException.java]: Error responses with this exception currently return a 400 status code.

If this method is removed, a 500 response will be returned instead. See `createEntityFromMigrationRequest()` and `createEntityFromRequest()` in `EntityRecordService`

It's fine to have this return a 500; however it needs to be replaced by a more suitable exception in `EntityRecordService`.

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/exception/SolrServiceException.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package eu.europeana.entitymanagement.solr.exception;
+
+public class SolrServiceException extends Exception {
```
Comment (ikattey) [inline entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/exception/SolrServiceException.java]: By convention, all custom exceptions should extend `EuropeanaApiException`

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrAgent.java [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+import org.apache.solr.client.solrj.beans.Field;
+
+import eu.europeana.entitymanagement.definitions.model.Agent;
+import eu.europeana.entitymanagement.vocabulary.AgentSolrFields;
+import eu.europeana.entitymanagement.vocabulary.EntitySolrFields;
+
+/*
+ * TODO:see how to save the referencedWebResource and isAggregatedBy fields for all entities
+ */
+public class SolrAgent extends Agent {
```
Comment (ikattey) [inline entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrAgent.java]: Consider making this a different type. See the implementation in `XmlAgentImpl`. 

This comment applies to all `Solr*` classes.

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/service/SolrService.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+public interface SolrService {
+
+	/**
+	 * This method stores objects in SOLR.
+	 * @param solrObject
+	 * @param doCommit commit
+	 * @throws SolrServiceException 
+	 */
+	public void storeEntity(Entity solrObject, boolean doCommit) throws SolrServiceException ;
```
Comment (ikattey) [inline entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/service/SolrService.java]: Method signature makes it very easy to store the wrong objects in Solr.

Consider introducing a new type for Solr, similar to what we do for XML handling.

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/service/SolrService.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package eu.europeana.entitymanagement.solr.service;
+
+import eu.europeana.entitymanagement.definitions.exceptions.UnsupportedEntityTypeException;
+import eu.europeana.entitymanagement.definitions.model.Entity;
+import eu.europeana.entitymanagement.solr.exception.SolrServiceException;
+
+public interface SolrService {
```
Comment (ikattey) [inline entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/service/SolrService.java]: The convention for this project is to avoid single-implementation interfaces (as mentioned on Slack).

Directly implement functionality in `SolrService`, instead of having `SolrService` interface and a `SolrServiceImpl` class.

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/service/impl/SolrServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+import eu.europeana.entitymanagement.solr.model.SolrConcept;
+import eu.europeana.entitymanagement.solr.model.SolrOrganization;
+import eu.europeana.entitymanagement.solr.model.SolrPlace;
+import eu.europeana.entitymanagement.solr.model.SolrTimespan;
+import eu.europeana.entitymanagement.solr.service.SolrService;
+import eu.europeana.entitymanagement.vocabulary.EntitySolrFields;
+import eu.europeana.entitymanagement.vocabulary.EntityTypes;
+
+@Service(AppConfigConstants.BEAN_EM_SOLR_SERVICE)
+public class SolrServiceImpl implements SolrService {
```
Comment (ikattey) [inline entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/service/impl/SolrServiceImpl.java]: As above, please implement functionality directly in `SolrService`.

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/SolrConfig.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+	}
+
+	private static final Logger logger = LogManager.getLogger(SolrConfig.class);
+
+	final EntityManagementConfiguration emConfiguration;
+	
+	@Bean
+	public SolrClient solrClient() {
+		logger.info("Configuring the solr client at the url: {}", emConfiguration.getSearchApiSolrUrl());
+		return new HttpSolrClient.Builder(emConfiguration.getSearchApiSolrUrl()).build();
```
Comment (ikattey) [inline entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/SolrConfig.java]: Solr instance for indexing entities is different from the Search API instance.

This should be a new instance / config property.

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/SolrConfig.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.solr.core.SolrTemplate;
+import org.springframework.data.solr.repository.config.EnableSolrRepositories;
+
+import eu.europeana.entitymanagement.common.config.EntityManagementConfiguration;
+
+@Configuration
+@EnableSolrRepositories
+public class SolrConfig {
```
Comment (ikattey) [inline entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/SolrConfig.java]: All SolrClient instances (including the two in `ScoringService`) should be initialised here as named beans.

**Review by ikattey — CHANGES_REQUESTED** [review]
@SrdjanStevanetic please see my comments below.

I did not check the field names, as @gsergiu is better placed to review those.

File: entity-management-definitions/src/main/java/eu/europeana/entitymanagement/definitions/exceptions/EntityCreationException.java (hunk 1 above)
Comment (SrdjanStevanetic) [reply entity-management-definitions/src/main/java/eu/europeana/entitymanagement/definitions/exceptions/EntityCreationException.java]: I reverted back the 400 status code.

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/exception/SolrServiceException.java (hunk 2 above)
Comment (SrdjanStevanetic) [reply entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/exception/SolrServiceException.java]: Improved.

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrAgent.java (hunk 3 above)
Comment (SrdjanStevanetic) [reply entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrAgent.java]: I think this is not necessary. We need to have the same fields in the Solr entities as in the main entity classes, and simply extending from the main classes keeps that validation in place. We also do not need to define the new fields in the Solr classes, simply use the @Field annotation on the setters for the existing fields. However, we need to define the base entity fields in all Solr classes independently and cannot keep them in one class, but it is fine I guess.

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/service/SolrService.java (hunk 4 above)
Comment (SrdjanStevanetic) [reply entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/service/SolrService.java]: Improved.

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/service/SolrService.java (hunk 5 above)
Comment (SrdjanStevanetic) [reply entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/service/SolrService.java]: Improved.

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/service/impl/SolrServiceImpl.java (hunk 6 above)
Comment (SrdjanStevanetic) [reply entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/service/impl/SolrServiceImpl.java]: Done.

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/SolrConfig.java (hunk 7 above)
Comment (SrdjanStevanetic) [reply entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/SolrConfig.java]: Improved.

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/SolrConfig.java (hunk 8 above)
Comment (SrdjanStevanetic) [reply entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/SolrConfig.java]: I have tried that and I think that Spring has some problems with using many beans with the same type even though I tried to distinguish them by using the bean names and the @Qualifier annotation while imorting them over @Autowired. Therefore I actually created the indexing SolrClient the same way as the other 2 solr clients in the ScoringService class.

Comment (ikattey) [reply entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/SolrConfig.java]: See [this commit](https://github.com/europeana/entity-management/compare/EA-2577-solr-indexing...EA-2577-solr-config).

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/service/impl/SolrServiceImpl.java (hunk 6 above)
Comment (ikattey) [reply entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/service/impl/SolrServiceImpl.java]: Good! I see the class is still in the `solr/service/impl` package however 🙂

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrAgent.java (hunk 3 above)
Comment (ikattey) [reply entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrAgent.java]: There's a lot of duplicated code in the Solr model classes, which can be eliminated with a different class hierarchy.

See [here](https://github.com/europeana/entity-management/compare/EA-2577-solr-indexing...EA-2577-solr-entities#diff-a0bb5be85e98ff1bec4215d9d8ba320a3b044302cccce6a5f1e6eda08a5250e8R11) for a demonstration of what I'm recommending.

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/SolrConfig.java (hunk 8 above)
Comment (SrdjanStevanetic) [reply entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/SolrConfig.java]: I see, the problem was in the SolrTemplate beans that did not work together with the SolrClient beans and you put the @Qualifier in the constructor. Fine.

File: entity-management-common/src/main/java/eu/europeana/entitymanagement/common/config/EntityManagementConfiguration.java [hunk 9]
```diff
@@ -28,14 +28,23 @@
...
     private String apiKeyUrl;
 
     @Value("${entitymanagement.solr.pr.url}")
     private String prSolrUrl;
 
     @Value("${entitymanagement.solr.searchapi.url}")
     private String searchApiSolrUrl;
+    
+    @Value("${entitymanagement.solr.searchapi.collection}")
+    private String searchApiSolrCollection;
```
Comment (gsergiu) [inline entity-management-common/src/main/java/eu/europeana/entitymanagement/common/config/EntityManagementConfiguration.java]: the collection of the search API is already part of the URL, this property is redundant

File: entity-management-common/src/main/java/eu/europeana/entitymanagement/common/config/EntityManagementConfiguration.java [hunk 10]
```diff
@@ -28,14 +28,23 @@
...
+    
+    @Value("${entitymanagement.solr.indexing.url}")
+    private String indexingSolrUrl;
+    
+    @Value("${entitymanagement.solr.indexing.collection}")
+    private String indexingSolrCollection;
```
Comment (gsergiu) [inline entity-management-common/src/main/java/eu/europeana/entitymanagement/common/config/EntityManagementConfiguration.java]: this property is redundant, the collection should be part of the solr URL

File: entity-management-solr/pom.xml [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+  </parent>
+
+  <groupId>eu.europeana.api</groupId>
+  <artifactId>entity-management-solr</artifactId>
+  <version>0.0.1-SNAPSHOT</version>
+
+  <name>entity-management-solr</name>
+  <description>A simple entity-management-solr.</description>
+  <!-- FIXME change it to the project's website -->
+  <url>http://www.example.com</url>
```
Comment (gsergiu) [inline entity-management-solr/pom.xml]: please remove this line

File: entity-management-solr/pom.xml [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+
+  <properties>
```
Comment (gsergiu) [inline entity-management-solr/pom.xml]: these properties are inherited from the base pom, please remove

File: entity-management-solr/pom.xml [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+	
+	<!-- https://mvnrepository.com/artifact/org.json/json -->
+	
+  </dependencies>
+
+  <build>
+   
+  </build>
+
+  <reporting>
```
Comment (gsergiu) [inline entity-management-solr/pom.xml]: do we need this plugin? If not, please remove

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrAgent.java [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+	private String dateOfTermination;
+
+	@Field(AgentSolrFields.GENDER)
+	private String gender;
+
+	public SolrAgent(Agent agent) {
+		super(agent);
+
+		
+		this.date = agent.getDate();
```
Comment (gsergiu) [inline entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrAgent.java]: aren't these values already set in the super(agent) ? Only the multilingual maps need preProcessing to append solrFieldName to labels, but that should also be applied automatically through the call of the supper implementation

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrConcept.java [hunk 15]
```diff
@@ -0,0 +1,199 @@
...
+public class SolrConcept extends Concept {
+
+	private String payload;
+	
+	public SolrConcept() {
+		super();
+	}
+
+	public SolrConcept(Concept concept) {
+		super();
```
Comment (gsergiu) [inline entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrConcept.java]: please switch to the use of super(concept) and remove the redundant calls to set the values.

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrEntity.java [hunk 16]
```diff
@@ -0,0 +1,142 @@
...
+import eu.europeana.entitymanagement.vocabulary.AgentSolrFields;
+import eu.europeana.entitymanagement.vocabulary.EntitySolrFields;
+import org.apache.solr.client.solrj.beans.Field;
+
+import java.util.List;
+import java.util.Map;
+
+public abstract class SolrEntity<T extends Entity> {
+
+    @Field(AgentSolrFields.TYPE)
```
Comment (gsergiu) [inline entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrEntity.java]: This is a common field, please use the EntitySolrFields class instead

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrEntity.java [hunk 17]
```diff
@@ -0,0 +1,142 @@
...
+    private String type;
+
+    @Field(AgentSolrFields.ID)
```
Comment (gsergiu) [inline entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrEntity.java]: This is a common field, please use the EntitySolrFields class insted

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrEntity.java [hunk 18]
```diff
@@ -0,0 +1,142 @@
+package eu.europeana.entitymanagement.solr.model;
+
+import eu.europeana.entitymanagement.definitions.model.Entity;
+import eu.europeana.entitymanagement.vocabulary.AgentSolrFields;
```
Comment (gsergiu) [inline entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrEntity.java:5]: Please remove this import and use the constants from EntitySolrFields class instead

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrOrganization.java [hunk 19]
```diff
@@ -0,0 +1,265 @@
...
+public class SolrOrganization extends Organization {
+
+	private String payload;
+	
+	public SolrOrganization() {
+		super();
+	}
+
+	public SolrOrganization(Organization organization) {
+		super();
```
Comment (gsergiu) [inline entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrOrganization.java]: use super(organization) and remove the use of redundant setters

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrPlace.java [hunk 20]
```diff
@@ -0,0 +1,168 @@
...
+public class SolrPlace extends Place {
+	
+	private String payload;
+
+	public SolrPlace() {
+		super();
+	}
+
+	public SolrPlace(Place place) {
+		super();
```
Comment (gsergiu) [inline entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrPlace.java]: use super(place) and remove the use of redundant setters

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrTimespan.java [hunk 21]
```diff
@@ -0,0 +1,146 @@
...
+public class SolrTimespan extends Timespan {
+	
+	private String payload;
+
+	public SolrTimespan() {
+		super();
+	}
+
+	public SolrTimespan(Timespan timespan) {
+		super();
```
Comment (gsergiu) [inline entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrTimespan.java]: use super(timespan) and remove the use of redundant setters

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/service/SolrService.java [hunk 22]
```diff
@@ -0,0 +1,186 @@
...
+import eu.europeana.entitymanagement.solr.exception.SolrServiceException;
+import eu.europeana.entitymanagement.vocabulary.EntitySolrFields;
+import eu.europeana.entitymanagement.vocabulary.EntityTypes;
+
+@Service(AppConfigConstants.BEAN_EM_SOLR_SERVICE)
+public class SolrService {
+
+	@Qualifier(AppConfigConstants.BEAN_JSON_MAPPER)
+	@Autowired
+	ObjectMapper emJsonMapper;
```
Comment (gsergiu) [inline entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/service/SolrService.java]: if this is the mapper used for serialization of the preview, please rename to payloadObjectMapper

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/utils/SolrEntityUtils.java [hunk 23]
```diff
@@ -0,0 +1,20 @@
+package eu.europeana.entitymanagement.solr.utils;
+
+import eu.europeana.entitymanagement.definitions.model.Agent;
+import eu.europeana.entitymanagement.definitions.model.Entity;
+import eu.europeana.entitymanagement.solr.model.SolrAgent;
+import eu.europeana.entitymanagement.solr.model.SolrEntity;
+
+public class SolrEntityUtils {
```
Comment (gsergiu) [inline entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/utils/SolrEntityUtils.java]: this class doesn't seem to be needed, if not used please remove

File: entity-management-solr/src/site/site.xml [hunk 24]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (gsergiu) [inline entity-management-solr/src/site/site.xml]: for what is this file needed?

File: entity-management-web/src/integration-test/java/eu/europeana/entitymanagement/solr/SolrServiceTest.java [hunk 25]
```diff
@@ -0,0 +1,68 @@
...
+    
+    @Qualifier(AppConfigConstants.BEAN_JSON_MAPPER)
+    @Autowired
+    private ObjectMapper objectMapper;
+    
+    @Autowired
+    JsonLdSerializer emJsonldSerializer;
+
+    //@Test
+    public void storeEntitiesInSolr() throws Exception {
```
Comment (gsergiu) [inline entity-management-web/src/integration-test/java/eu/europeana/entitymanagement/solr/SolrServiceTest.java]: please create one test for each entity class

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/service/SolrService.java [hunk 26]
```diff
@@ -0,0 +1,186 @@
...
+	@Autowired 
+    public SolrService(EntityManagementConfiguration emConfig) {
+		emConfiguration=emConfig;
+    	indexingSolrClient = new HttpSolrClient.Builder(emConfiguration.getIndexingSolrUrl()).build();
+    }
+
+	public void storeEntity(final SolrEntity<? extends Entity> solrEntity, boolean doCommit) throws SolrServiceException {
+
+		// TODO: adapt and uncomment
+//			try {
```
Comment (gsergiu) [inline entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/service/SolrService.java]: this is that EA-2578 has to implement.

**Review by gsergiu — CHANGES_REQUESTED** [review]
please implement the required changes and complete serialization of payload field

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrAgent.java (hunk 3 above)
Comment (SrdjanStevanetic) [reply entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrAgent.java]: Improved.

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/service/impl/SolrServiceImpl.java (hunk 6 above)
Comment (SrdjanStevanetic) [reply entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/service/impl/SolrServiceImpl.java]: Improved.

File: entity-management-common/src/main/java/eu/europeana/entitymanagement/common/config/EntityManagementConfiguration.java (hunk 9 above)
Comment (SrdjanStevanetic) [reply entity-management-common/src/main/java/eu/europeana/entitymanagement/common/config/EntityManagementConfiguration.java]: Removed.

File: entity-management-common/src/main/java/eu/europeana/entitymanagement/common/config/EntityManagementConfiguration.java (hunk 10 above)
Comment (SrdjanStevanetic) [reply entity-management-common/src/main/java/eu/europeana/entitymanagement/common/config/EntityManagementConfiguration.java]: Removed.

File: entity-management-solr/pom.xml (hunk 11 above)
Comment (SrdjanStevanetic) [reply entity-management-solr/pom.xml]: Removed.

File: entity-management-solr/pom.xml (hunk 12 above)
Comment (SrdjanStevanetic) [reply entity-management-solr/pom.xml]: Removed.

File: entity-management-solr/pom.xml (hunk 13 above)
Comment (SrdjanStevanetic) [reply entity-management-solr/pom.xml]: Removed.

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrAgent.java (hunk 14 above)
Comment (SrdjanStevanetic) [reply entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrAgent.java]: The new solr classes are implemented.

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrConcept.java (hunk 15 above)
Comment (SrdjanStevanetic) [reply entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrConcept.java]: Improved.

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrEntity.java (hunk 16 above)
Comment (SrdjanStevanetic) [reply entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrEntity.java]: Done.

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrEntity.java (hunk 17 above)
Comment (SrdjanStevanetic) [reply entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrEntity.java]: Done.

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrEntity.java (hunk 18 above)
Comment (SrdjanStevanetic) [reply entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrEntity.java:5]: Done.

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrOrganization.java (hunk 19 above)
Comment (SrdjanStevanetic) [reply entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrOrganization.java]: Done.

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrPlace.java (hunk 20 above)
Comment (SrdjanStevanetic) [reply entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrPlace.java]: Done.

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrTimespan.java (hunk 21 above)
Comment (SrdjanStevanetic) [reply entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrTimespan.java]: Done.

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/service/SolrService.java (hunk 22 above)
Comment (SrdjanStevanetic) [reply entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/service/SolrService.java]: Done.

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/utils/SolrEntityUtils.java (hunk 23 above)
Comment (SrdjanStevanetic) [reply entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/utils/SolrEntityUtils.java]: Removed.

File: entity-management-solr/src/site/site.xml (hunk 24 above)
Comment (SrdjanStevanetic) [reply entity-management-solr/src/site/site.xml]: Removed.

File: entity-management-web/src/integration-test/java/eu/europeana/entitymanagement/solr/SolrServiceTest.java (hunk 25 above)
Comment (SrdjanStevanetic) [reply entity-management-web/src/integration-test/java/eu/europeana/entitymanagement/solr/SolrServiceTest.java]: Done.

File: entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/service/SolrService.java (hunk 26 above)
Comment (SrdjanStevanetic) [reply entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/service/SolrService.java]: Ok.
